orion/sites/admin/src/location/location.js

```diff
@@ -20,29 +24,97 @@ export class LocationPage extends React.Component {
+        } else {
+            this.setState({ numSelected: this.state.numSelected - 1 });
+        }
+    }
+
+    onClickSelectAll() {
+        var items = document.getElementsByName("unpublished");
+        for (var i = 0; i < items.length; i++) {
+            if (items[i].type == "checkbox") {
+                items[i].checked = true;
+            }
+        }
+        this.setState({ numSelected: items.length });
+    }
+
+    onClickPublish() {
+        console.log("clicked publish");
+    }
+
     render() {
         const location = this.state.list.map((location, index) => {
-            return <LocationRow key={index} location={location} />;
+            return (
+                <LocationRow
+                    key={index}
+                    location={location}
+                    onCheckRow={this.onCheckRow}
+                />
+            );
         });
         const numLocations = location.length;
         return (
             <div id="view-location">
-                <h1>All Locations ({numLocations})</h1>
+                <div>
+                    <h1>All Locations ({numLocations})</h1>
+                    <p>
+                        You have {this.state.numUnpublished} unpublished items.{" "}
+                        <br />
+                        You have selected {this.state.numSelected} items to
+                        publish.
+                    </p>
+                </div>
                 <ul id="list-heading">
+                    <button
+                        className="li-small"
+                        onClick={this.onClickSelectAll}>
+                        Select All
+                    </button>
                     <li className="li-med">Location ID</li>
                     <li className="li-large">Address</li>
                     <li className="li-large">Room</li>
                 </ul>
                 <ul>{location}</ul>
-                <button id="add-location">
-                    <Link to={"/locations/add"}>Add Location</Link>
-                </button>
+                <div id="list-buttons">
+                    <button>
+                        <Link to={"/locations/add"} id="add-location">
+                            Add Location
+                        </Link>
+                    </button>
+                    <button
+                        id="publish"
+                        className="publish"
+                        onClick={this.onClickPublish}>
+                        Publish
+                    </button>
+                </div>
             </div>
         );
     }
 }
 
 class LocationRow extends React.Component {
+    renderCheckbox(isUnpublished) {
+        let checkbox = <div> </div>;
+        if (isUnpublished) {
+            return (checkbox = (
+                <input
+                    className="li-small"
+                    type="checkbox"
+                    name="unpublished"
+                    id="unpublished"
```

Comment:
I think this might be the problem. id's are used for main sections and are SUPPOSED to only be for elements that appear once per page. This is a checkbox, so you'll have many of them - you want to use a className.

After that, when you search for all checkboxes, I think you can search by Class instead of search by Id.